src/idpi/grib_decoder.py

```diff
@@ -1,26 +1,134 @@
 """Decoder for grib data."""
-# Third-party
-import cfgrib  # type: ignore
+# Standard library
+import os
+from contextlib import contextmanager
+from pathlib import Path
 
+# Third-party
+import earthkit.data  # type: ignore
+import eccodes  # type: ignore
+import numpy as np
+import xarray as xr
 
-def load_data(outds, fields, datafile, chunk_size=10):
-    chunk_arg = {}
-    if chunk_size:
-        chunk_arg = {"chunks": {"generalVerticalLayer": chunk_size}}
 
-    # Note: dataset assignment is based on typeOfLevel in cfgrib
-    dss = cfgrib.open_datasets(
-        datafile,
-        backend_kwargs={"read_keys": ["typeOfLevel", "gridType"]},
-        encode_cf=("time", "geography", "vertical"),
-        **chunk_arg,
+@contextmanager
+def cosmo_grib_defs():
```

Comment:
this is a good change. I think this potentially might eliminate the need to use pytest markers: 
https://github.com/MeteoSwiss-APN/icon_data_processing_incubator/blob/main/pytest.ini